src/System.Linq/src/System/Linq/Skip.SpeedOpt.cs

```diff
@@ -12,5 +13,30 @@ public static partial class Enumerable
             source is IList<TSource> sourceList ?
                 (IEnumerable<TSource>)new ListPartition<TSource>(sourceList, count, int.MaxValue) :
                 new EnumerablePartition<TSource>(source, count, -1);
+
+        private static IEnumerable<TSource> SkipLastEnumerableFactory<TSource>(IEnumerable<TSource> source, int count)
+        {
+            Debug.Assert(count > 0);
+
+            if (source is IPartition<TSource> partition)
+            {
+                int length = partition.GetCount(true);
```

Comment:
Nit, possibly `int length = partition.GetCount(onlyIfCheap: true)`

Reply:
@danmosemsft  Thanks for your review. I will push a commit to add the parameter name.